src/models/tabularmodel.jl

```diff
@@ -0,0 +1,141 @@
+"""
+    emb_sz_rule(n_cat)
+
+Returns an embedding size corresponding to the number of classes for a 
+categorical variable using the rule of thumb present in python fastai.
+(see https://github.com/fastai/fastai/blob/2742fe844573d06e700f869839fb9ec5f3a9bca9/fastai/tabular/model.py#L12)
+"""
+function emb_sz_rule(n_cat)
+    min(600, round(1.6 * n_cat^0.56))
+end
+
+"""
+    get_emb_sz(cardinalities, [size_overrides])
+    get_emb_sz(cardinalities; catcols, [size_overrides])
+
+Returns a collection of tuples containing embedding dimensions corresponding to 
+number of classes in categorical columns present in `cardinalities` and adjusting for nans. 
+
+## Keyword arguments
+
+- `size_overrides`: Depending on the method used, this could either be a collection of 
+    Integers and `nothing` or an indexable collection with column name as key and size
+    to override it with as the value. In the first case, the integer present at any index 
+    will be used to override the rule of thumb for getting embedding sizes.
+- `categorical_cols`: A collection of categorical column names.
+"""
+
+function get_emb_sz(cardinalities, size_overrides=fill(nothing, length(cardinalities)))
+    map(Iterators.enumerate(cardinalities)) do (i, cardinality)
+        emb_dim = isnothing(size_overrides[i]) ? emb_sz_rule(cardinality+1) : size_overrides[i]
+        (Int64(cardinality)+1, Int64(emb_dim))
+    end
+end
+
+function get_emb_sz(cardinalities; catcols, size_overrides=Dict())
+    keylist = keys(size_overrides)
+    overrides = map(catcols) do col
+        col in keylist ? size_overrides[col] : nothing
+    end
+    get_emb_sz(cardinalities, overrides)
+end
+
+function sigmoidrange(x, low, high)
+    @. Flux.sigmoid(x) * (high - low) + low
+end
+
+function tabular_embedding_backbone(embedding_sizes, dropoutprob=0.)
+    embedslist = [Flux.Embedding(ni, nf) for (ni, nf) in embedding_sizes]
+    emb_drop = dropoutprob==0. ? identity : Dropout(dropoutprob)
+    Chain(
+        x -> tuple(eachrow(x)...), 
+        Parallel(vcat, embedslist), 
+        emb_drop
+    )
+end
```

Comment:
```suggestion
function tabular_embedding_backbone(embedding_sizes, dropout_rates=0.)
    embedslist = [Flux.Embedding(ni, nf) for (ni, nf) in embedding_sizes]
    emb_drop = iszero(dropout_rates) ? identity : Dropout(dropout_rates)
    Chain(
        x -> tuple(eachrow(x)...), 
        Parallel(vcat, embedslist), 
        emb_drop
    )
end
```

Reply:
`emb_drop` could even be further inlined if desired. If preserving model structure is more important than a bit of lost performance, then passing `active=iszero(dropout_rates)` instead of using a ternary is also a valid option.

Reply:
Is `active` an argument to something? How would that work?

Reply:
`active` is the third arg to `BatchNorm`, yes. I wouldn't worry about that now though, it's a minor tweak we can always revisit later.